aggregateur/validators.py

```diff
@@ -342,3 +342,86 @@ def schema_json_data(self):
         with open(self.filepath(self.SCHEMA_FILENAME)) as f:
             self.schema_data = json.load(f)
         return self.schema_data
+
+
+class GenericValidator(BaseValidator):
+    SCHEMA_FILENAME = "schema.yml"
+
+    def __init__(self, repo):
+        super().__init__(repo)
+        self.schema_data = None
+        self.has_changelog = False
+
+    def schemas_metadata(self):
+        return [
+            {
+                "path": self.SCHEMA_FILENAME,
+                "original_path": self.SCHEMA_FILENAME,
+                "title": self.title,
+                "latest_url": self.schema_url(self.SCHEMA_FILENAME),
+            }
+        ]
+
+    @property
+    def title(self):
+        return self.get_schema_data()["title"]
+
+    @property
+    def description(self):
+        return self.get_schema_data()["description"]
+
+    @property
+    def homepage(self):
+        return self.get_schema_data()["homepage"]
+
+    def validate(self):
+        super().validate()
+        self.check_file_exists(self.SCHEMA_FILENAME)
+        self.check_schema(self.SCHEMA_FILENAME)
+        self.check_extra_keys()
+
+    def extract(self):
+        files = {
+            self.SCHEMA_FILENAME: self.filepath(self.SCHEMA_FILENAME),
+            "README.md": self.filepath_or_none("README.md"),
+            "SEE_ALSO.md": self.filepath_or_none("SEE_ALSO.md"),
+            "CONTEXT.md": self.filepath_or_none("CONTEXT.md"),
+        }
+
+        if self.is_latest_version():
+            changelog_path = self.filepath_or_none(self.CHANGELOG_FILENAME)
+            files[self.CHANGELOG_FILENAME] = changelog_path
+            self.has_changelog = changelog_path is not None
+
+        self.move_files(files)
+
+    def check_extra_keys(self):
+        keys = ["title", "description", "homepage", "version"]
+        for key in [k for k in keys if k not in self.get_schema_data()]:
+            message = "Key `%s` is a required key and is missing from %s" % (
+                key,
+                self.SCHEMA_FILENAME,
+            )
+            raise exceptions.InvalidSchemaException(self.repo, message)
+
+    def open_schema_file(self):
+        data = None
+        with open(self.filepath(self.SCHEMA_FILENAME)) as f:
+            data = yaml.safe_load(f)
+        return data
+
+    def check_schema(self, filename):
+        try:
+            self.open_schema_file()
+        except yaml.error.YAMLError as e:
```

Comment:
Is this the only exception that can be raised here? If the file doesn't exist, `open` will raise something else. I think this should never happen because `validate` checks for the existence first